src/osemgrep/reporting/Sarif_output.ml

```diff
@@ -143,7 +143,11 @@ let rule ~(hide_nudge : bool) (ctx : Out.format_context) (rule : Rule.t) :
     | Some (JSON.String source) -> Some source
     | Some _
     | None ->
-        None
+        match JSON.member "source-rule-uri" metadata with
```

Comment:
This is wrong its source-rule-url.

Reply:
ok, so this change does what we want. We'll document the `metadata` section of Semgrep rules, eventually.

Please add a comment explaining the following since it's not documented:
```
(*
  In a Semgrep rule's metadata section, two fields may provide URLs:
  - source: populated dynamically by the Semgrep registry serving the rule, it's a URL that offers information about the rule.
  - source-rule-url: optional string, a URL for the source of inspiration for the rule.

  The SARIF format supports only one URL under the field 'helpUri'. Semgrep populates it with metadata.source if available, metadata.source-rule-url otherwise as a fallback.
*)
```

Reply:
I changed it to `source-rule-url` (oops) and added the comment.